### PR TITLE
fix publish_module_bundle

### PR DESCRIPTION
**Pull request by nkysg**
fix publish_module_bundle
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: abi/resolver/src/lib.rs [hunk 1]
```diff
@@ -127,9 +135,14 @@ impl<'a> ABIResolver<'a> {
             .resolver
             .get_module(module_id.address(), module_id.name())?;
         let struct_def = find_struct_def_in_module(module.as_ref(), name)?;
-        let (name, s) = Struct::new(&module, module.struct_def_at(struct_def));
+        let m = Struct::new(&module, module.struct_def_at(struct_def));
+        let (name, s) = match m {
+            Ok(m) => (m.0, m.1),
+            e => return Err(anyhow!("get module {} {:?} error", module_id, e)),
```
Comment (jolestar) [inline abi/resolver/src/lib.rs:141]: 同上

File: abi/resolver/src/lib.rs [hunk 2]
```diff
@@ -54,6 +54,10 @@ impl<'a> ABIResolver<'a> {
 
     fn resolve_compiled_module(&self, module: &CompiledModule) -> Result<ModuleABI> {
         let m = Module::new(module);
+        let m = match m {
+            Ok(m) => m,
+            e => return Err(anyhow!("get module {:?} {:?} error", module, e)),
+        };
```
Comment (jolestar) [inline abi/resolver/src/lib.rs:60]: 这种是不是 `Module::new(module)?` 就可以

**Review by jolestar — APPROVED** [review]

File: abi/resolver/src/lib.rs (hunk 2 above)
Comment (nkysg) [reply abi/resolver/src/lib.rs:60]: 这里是因为Module::new返回是Result<T, VMError>, 上面函数返回Result<()>, 直接？回报错

File: abi/resolver/src/lib.rs (hunk 1 above)
Comment (nkysg) [reply abi/resolver/src/lib.rs:141]: 这里是因为Module::new返回是Result<T, VMError>, 上面函数返回Result<()>, 直接？回报错

File: abi/resolver/src/lib.rs (hunk 2 above)
Comment (jolestar) [reply abi/resolver/src/lib.rs:60]: 哦，那应该用 `map_err` 转换一下

Comment (nkysg) [reply abi/resolver/src/lib.rs:60]: 我修改下

**Review by jolestar — APPROVED** [review]

**Pull request merged**